src/scalingengine/server/scaling_handler.go

```diff
@@ -60,39 +66,74 @@ func (h *ScalingHandler) HandleScale(w http.ResponseWriter, r *http.Request, var
 func (h *ScalingHandler) Scale(appId string, trigger *models.Trigger) (int, error) {
 	logger := h.logger.WithData(lager.Data{"appId": appId})
```

Comment:
One trick we can do is set a defer block to do the SaveScalingHistory and just do the setup throughout the code. This way you don't have to have the SaveScalingHistory scattered everywhere especially when you always want it to occur.